embassy-stm32/src/rcc/f013.rs

```diff
@@ -187,21 +234,39 @@ pub(crate) unsafe fn init(config: Config) {
                 }
                 (Pllsrc::HSI_DIV2, unwrap!(hsi))
             }
+            #[cfg(not(stm32f107))]
             PllSource::HSE => (Pllsrc::HSE_DIV_PREDIV, unwrap!(hse)),
+            #[cfg(stm32f107)]
+            PllSource::HSE => (
```

Comment:
this is a bit of a lie, if you set `PllSource::HSE` the source can end up being either HSE or PLL2. I'd suggest adding a new variant `PllSource::PLL2` instead, and removing the `prediv1_src` field from config.